samples/excel/85-preview-apis/data-types-custom-functions.yaml

```diff
@@ -0,0 +1,334 @@
+order: 5
+id: excel-data-types-custom-functions
+name: 'Data types: Custom functions'
+description: This sample shows how to write custom functions that return entity value data types.
+host: EXCEL
+api_set:
+    ExcelApi: BETA (PREVIEW ONLY)
+script:
+    content: |
+        /**
+         * Search for products that match a given substring. Try =SCRIPTLAB.DATATYPESCUSTOMFUNCTIONS.PRODUCTSEARCH("syrup").
+         * @customfunction
+         * @param search The string to search for in the sample JSON data.
+         * @param [lookAt] Define whether the search should be a match of the whole product name or part of the product name. If omitted, set to lookAt = "Part".
+         * @return {Promise<any[][]>} Search results as one or more data type entity values.
+         */
+        async function productSearch(query: string, lookAt?: string): Promise<any[][]> {
```

Comment:
It's not clear to me what else `lookAt` could be. Should this just be a boolean value like `completeMatch` in [SearchCriteria](https://docs.microsoft.com/en-us/javascript/api/excel/excel.searchcriteria?view=excel-js-preview#excel-excel-searchcriteria-completematch-member)?

Reply:
@jakobpn Can you weigh in on this?

Reply:
Agree. That is not particularly well documented in the signature of the function. I would suggest adding the 2 possible options to the @param comment for lookAt. I think I called this "lookAt" to be similar to the find methods in the Office JS API.

Reply:
Thanks for the feedback. I'll change this param to a boolean called completeMatch so that the options (true/false) are more obvious.